src/Evolution/Systems/Cce/Actions/InitializeWorldtubeBoundary.hpp

```diff
@@ -9,30 +9,89 @@
 
 #include "DataStructures/DataBox/DataBox.hpp"
 #include "DataStructures/VariablesTag.hpp"
+#include "Evolution/Systems/Cce/AnalyticBoundaryDataManager.hpp"
 #include "Evolution/Systems/Cce/BoundaryData.hpp"
 #include "Evolution/Systems/Cce/InterfaceManagers/GhInterfaceManager.hpp"
+#include "Evolution/Systems/Cce/InterfaceManagers/GhLocalTimeStepping.hpp"
 #include "Evolution/Systems/Cce/InterfaceManagers/GhLockstep.hpp"
 #include "Evolution/Systems/Cce/OptionTags.hpp"
 #include "Evolution/Systems/Cce/Tags.hpp"
-#include "Evolution/Systems/Cce/WorldtubeDataManager.hpp"
 #include "NumericalAlgorithms/Interpolation/SpanInterpolator.hpp"
 #include "Parallel/Info.hpp"
 #include "Parallel/Invoke.hpp"
 #include "ParallelAlgorithms/Initialization/MutateAssign.hpp"
 #include "Utilities/Gsl.hpp"
+#include "Utilities/MakeString.hpp"
 #include "Utilities/Requires.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
 namespace Cce {
+/// \cond
+template <typename Metavariables>
+struct H5WorldtubeBoundary;
+template <typename Metavariables>
+struct AnalyticWorldtubeBoundary;
+template <typename Metavariables>
+struct GhWorldtubeBoundary;
+/// \endcond
 namespace Actions {
+
+namespace detail {
+template <typename Initializer, typename ManagerTag,
+          typename BoundaryCommunicationTagsList>
+struct InitializeWorldtubeBoundaryBase {
+  using initialization_tags = tmpl::list<ManagerTag>;
+  using initialization_tags_to_keep = tmpl::list<ManagerTag>;
+  using const_global_cache_tags = tmpl::list<Tags::LMax>;
+
+  using simple_tags =
+      tmpl::list<::Tags::Variables<BoundaryCommunicationTagsList>>;
+
+  template <
+      typename DataBoxTagsList, typename... InboxTags, typename ArrayIndex,
+      typename Metavariables, typename ActionList, typename ParallelComponent,
+      Requires<tmpl::list_contains_v<DataBoxTagsList, ManagerTag>> = nullptr>
```

Comment:
I think this `Requires` is not needed because you have the `if constexpr` in the function below.